pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -422,12 +437,36 @@ func (b *MasqueradeLibvirtSpecGenerator) generate(domainIface api.Interface) err
 	return nil
 }
 
+func (b *MasqueradeLibvirtSpecGenerator) decorateDomainIfaceSpec(domainIface *api.Interface) error {
+	podNicLink, err := b.handler.LinkByName(b.podInterfaceName)
```

Comment:
I have to ask how is this better ? I mean... We were reading the info from a file. Now we read from netlink. How's that any better ? 

I actually think the best thing to do would be to remove all netlink dependencies from phase#2, and let it just process already available data into the dhcp config + api interface (domxml). That would at least have the benefit of dropping dependencies from phase#2, and allowing it to be extremely simple - to test.

Reply:
Thanks for elaborating. Reducing the amount of data being passed around will make explicit exactly which data cannot be calculated on phase#2 (i.e. which data must be calculated on phase#1 then passed between processes).